README.md

```diff
@@ -1,219 +1,251 @@
-use MySQL for this purpose. If you don't have an available MySQL service, you can start one up using docker:
-`docker run -e "MYSQL_ROOT_PASSWORD=root" -p 127.0.0.1:3306:3306 --name=atomfeed-db -d mysql/mysql-server:latest`
-
-Now you should have MySQL running on the default port 3306, and can run:
-
-`mysql --user=USER --password=PASSWORD < utils/dbdump/create_db.sql`
-
-This will create a database called `atomfeed_client` with required tables (the
-`USER` should have permission to create databases). If you want to change the
-default database name `atomfeed_client`, you can edit [`utils/dbdump/create_db.sql`](
-utils/dbdump/create_db.sql) but then you need to change the database name in
-[`src/main/resources/hibernate.default.properties`](
-src/main/resources/hibernate.default.properties) accordingly.
-
-## Create the sink FHIR store and BigQuery dataset
-To set up GCP project that you can use as a sink FHIR store:
+## Prerequisites
+
+-   An [OpenMRS](https://openmrs.org) instance with the latest version of the
+    [FHIR2 Module](https://addons.openmrs.org/show/org.openmrs.module.openmrs-fhir2-module)
+    installed.
+    -   There is a [OpenMRS Reference Application](#run-openmrs-using-docker)
+        image with these prerequisites and demo data you can use to try things
+        out.
+-   A target output for the data. Supported options are Apache Parquet files
+    (not supported by Atom Feed mode), FHIR API resources, and
+    [Google Cloud Platform FHIR stores](https://cloud.google.com/healthcare/docs/how-tos/fhir).
+    -   You can use our [HAPI FHIR server](#run-hapi-fhir-server-using-docker)
+        image for testing FHIR API targets.
+    -   [Learn how to create a compatible GCP FHIR store](#create-a-google-cloud-fhir-store-and-bigquery-dataset).
+
+### Installing the FHIR2 Module
+
+**To install via web**: Log in to OpenMRS as an administrator, go to **System
+Administration > Manage Modules > Search from Addons**, search for the `FHIR2`
+module, and install. Make sure to install the FHIR2 module, not FHIR.
+
+**To install via .omod file**: Go to
+[FHIR2 Module](https://addons.openmrs.org/show/org.openmrs.module.openmrs-fhir2-module)
+and download the .omod file. Go to **Advanced Administration > Module
+Properties** and find the modules folder. Move the .omod file to the modules
+folder then restart OpenMRS.
+
+[Learn more about OpenMRS modules](https://guide.openmrs.org/en/Configuration/customizing-openmrs-with-plug-in-modules.html).
+
+## Setup
+
+1.  [Install the FHIR2 Module](#installing-the-fhir2-module) if necessary.
+2.  [Clone](https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/cloning-a-repository)
+    the
+    [OpenMRS FHIR Analytics](https://github.com/GoogleCloudPlatform/openmrs-fhir-analytics)
+    project to your machine.
+3.  Set the `utils` directory to world-readable: `chmod -r 755 ./utils`.
```

Comment:
I think this will make the content of `utils/` executable too which is not a great. Only `utils` and `utils/dbdump` need to be made executable (and everything world-readable).

Reply:
I can't find a particularly elegant way to set directories to 755 but files to 644. The best I found was this two-liner.

```
find ./utils -type d -exec chmod 755 {} \;
find ./utils -type f -exec chmod 644 {} \;
```

Do you have a better command?

Reply:
This approach is okay but the second command will remove the execute bit from those that have it. Another approach is to be more specific, e.g.,:
```
$ chmod a+rx ./utils ./utils/dbdump
$ chmod -R a+r ./utils
```